### PR TITLE
Implement the fusion of convolution 3D and bias for mkldnn

### PR DESCRIPTION
**Pull request by yihuaxu**
According to the performance status of DAM model, implemented the pass of convolution 3D and bias for mkldnn to accelerate the data processing.

**Platform**:  Intel(R) Xeon(R) CPU E5-2699 v3 @ 2.30GHz
**Model Path**: third_party/inference_demo/small_dam/model
**Batch Size**: 300
**Command**: ./paddle/fluid/inference/tests/api/test_analyzer_small_dam --infer_model=third_party/inference_demo/small_dam/model/ --infer_data=third_party/inference_demo/small_dam/data.txt --gtest_filter=Analyzer_dam.profile_mkldnn --paddle_num_threads=1 --repeat=1 --batch_size=300 --max_turn_num=1 --test_all_data --profile
**Data Source**: third_party/inference_demo/small_dam/data.txt.

Dependency PR: [14707](https://github.com/PaddlePaddle/Paddle/pull/14707)

The following is the comparison with the different scenarios.

image

### CODE REVIEW

**Comment by yihuaxu** [comment]
@luotao1 @tensor-tang @tpatejko Please help us review it. Thanks a lot!

**Review by wojtuss — DISMISSED** [review]

File: paddle/fluid/framework/ir/conv3d_bias_mkldnn_fuse_pass.cc [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "paddle/fluid/framework/ir/conv3d_bias_mkldnn_fuse_pass.h"
+
+REGISTER_PASS(conv3d_bias_mkldnn_fuse_pass,
```
Comment (luotao1) [inline paddle/fluid/framework/ir/conv3d_bias_mkldnn_fuse_pass.cc]: You can `REGISTER_PASS(conv3d_bias_mkldnn_fuse_pass)` in `conv_bias_mkldnn_fuse_pass.cc`. And don't add `conv3d_bias_mkldnn_fuse_pass.h/cc`

File: paddle/fluid/operators/activation_mkldnn_op.cc [hunk 2]
```diff
@@ -100,9 +100,6 @@ void eltwise_forward(const framework::ExecutionContext &ctx,
   const T *x_data = x->data<T>();
   T *y_data = y->mutable_data<T>(ctx.GetPlace());
 
-  PADDLE_ENFORCE(x->dims().size() == 2 || x->dims().size() == 4,
-                 "Input dim must be with 2 or 4");
-
```
Comment (luotao1) [inline paddle/fluid/operators/activation_mkldnn_op.cc]: why remove the `PADDLE_ENFORCE` in this PR?

File: paddle/fluid/framework/ir/conv3d_bias_mkldnn_fuse_pass.cc (hunk 1 above)
Comment (yihuaxu) [reply paddle/fluid/framework/ir/conv3d_bias_mkldnn_fuse_pass.cc]: These files was already merged into conv_bias_mkldnn_fuse_pass.cc.

File: paddle/fluid/operators/activation_mkldnn_op.cc (hunk 2 above)
Comment (yihuaxu) [reply paddle/fluid/operators/activation_mkldnn_op.cc]: As can not verify dims be 5 and others, so I add the value checking when the dims equal 3.

**Comment by luotao1** [comment]
This PR also depends on #14768

File: paddle/fluid/framework/ir/graph_pattern_detector.cc [hunk 3]
```diff
@@ -1030,23 +1030,26 @@ PDNode *patterns::ElewiseAddActInplaceGrad::operator()(
 }
 
 PDNode *patterns::ConvBias::operator()(
-    paddle::framework::ir::PDNode *conv_input) {
+    paddle::framework::ir::PDNode *conv_input, bool is_conv3d) {
```
Comment (luotao1) [inline paddle/fluid/framework/ir/graph_pattern_detector.cc:1033]:
```
std::string type = is_conv3d() ? "conv3d" : "conv2d";
```

Comment (yihuaxu) [reply paddle/fluid/framework/ir/graph_pattern_detector.cc:1033]: Done! This PR is not associated with [PR#14768](https://github.com/PaddlePaddle/Paddle/pull/14768).

Comment (yihuaxu) [reply paddle/fluid/framework/ir/graph_pattern_detector.cc:1033]: You can run "Analyzer_dam.compare_mkldnn" and check the below.
--- Running IR pass [**conv3d_bias_mkldnn_fuse_pass**]
---  **detect 2 subgraphs**
--- Running IR pass [conv_relu_mkldnn_fuse_pass]

**Comment by luotao1** [comment]
ON `Intel(R) Xeon(R) CPU E5-2620 v3 @ 2.40GHz`
- command
```
/paddle/fluid/inference/tests/api/test_analyzer_small_dam --infer_model=third_party/inference_demo/small_dam/model/ --infer_data=third_party/inference_demo/small_dam/data.txt --gtest_filter=Analyzer_dam.profile_mkldnn --paddle_num_threads=1 --repeat=1 --batch_size=1 --max_turn_num=1 --test_all_data --profile
```
- conv3d: 
   - before: 1369.79ms (mkl version)
   - after: 565.264ms (mkldnn version)
- whole small-dam network slows in mkldnn version, since #14768 isn't ready now.

**Review by luotao1 — APPROVED** [review]
LGTM!

**Pull request merged**